src/Core/Checkout/Document/Event/DocumentOrderCriteriaEvent.php

```diff
@@ -0,0 +1,48 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Checkout\Document\Event;
+
+use Shopware\Core\Checkout\Document\Renderer\DocumentRendererConfig;
+use Shopware\Core\Checkout\Document\Struct\DocumentGenerateOperation;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\Log\Package;
+use Symfony\Contracts\EventDispatcher\Event;
+
+#[Package('checkout')]
+abstract class DocumentOrderCriteriaEvent extends Event
```

Comment:
I think, we should get rid of these empty event classes with no benefit.
Make this class not abstract and add a name. Something like `Shopware\Core\Framework\DataAbstractionLayer\Event\EntityLoadedEvent`.
Use `Shopware\Core\Framework\Event\GenericEvent` interface
```php
    public function __construct(
        private readonly Criteria $criteria,
        private readonly Context $context,
        private readonly array $operations,
        private readonly DocumentRendererConfig $documentRendererConfig,
        string $rendererType
    ) {
        $this->name = $rendererType . '.document.criteria';
    }
````

Reply:
hey, sorry for the late reply, I was quite ill the last week, okay I will look into this, I thought it would be nicer, to have explicit classes to subscribe for different criteria events and I did it similar to the result events.

But I will have look how to optimize this.